dask/tests/test_base.py

```diff
@@ -224,6 +226,49 @@ def test_normalize_base():
         assert normalize_token(i) is i
 
 
+def test_normalize_object():
```

Comment:
This test can be simplified a bit. In particular, we can probably just test that an informative error is raised when `tokenize.allow-random` is `False` and we attempt to `tokenize` an object which can't be deterministically hashed. See this comment https://github.com/dask/dask/pull/7413/files#r596499598 for an outline of a simplified test.

Reply:
Hmm.. my concerns here are that, without tests proving that true negatives are handled as such, we may not have sufficient confidence that the newly introduced functionality doesn't break behavior which it shouldn't affect. And, somehow, I don't feel comfortable to rely on other tests, external to this one, to do this for us.

Of course, if you strongly believe we could make do without the extra set of tests, I'll have to remove them (no issues about that). Just wanted to share my concerns.

Reply:
> This is a great point -- if you want to add a check for `tokenize.allow-random` in those locations too, that would be welcome

Hi @jrbourbeau - yes, I've opted to implement configurable behavior for those as well. The commits have already been added to this PR.

Reply:
> This test can be simplified a bit.

I've made an attempt at simplifying the tests. Hopefully, looking better from your point of view now. Let me know, please, if you'd like anything else changed.